modules/alloy/src/main/ts/ephox/alloy/api/ui/FloatingToolbarButton.ts

```diff
@@ -57,111 +88,99 @@ const setGroups = (button: AlloyComponent, toolbar: AlloyComponent, detail: Floa
   Toggling.on(button);
 };
 
-const makeSandbox = (button: AlloyComponent, spec: FloatingToolbarButtonSpec, detail: FloatingToolbarButtonDetail) => {
-  const ariaControls = AriaControls.manager();
-
-  const onOpen = (sandbox: AlloyComponent, toolbar: AlloyComponent) => {
-    detail.fetch().get((groups) => {
-      setGroups(button, toolbar, detail, spec.layouts, groups);
-      ariaControls.link(button.element);
-      Keying.focusIn(toolbar);
-    });
-  };
+const makeSandbox = (
+  button: AlloyComponent,
+  spec: FloatingToolbarButtonSpec,
+  detail: FloatingToolbarButtonDetail,
+  ariaControls: AriaControls.AriaManager
+) => ({
+  dom: {
+    tag: 'div',
+    attributes: {
+      id: ariaControls.id
+    }
+  },
+  behaviours: Behaviour.derive(
+    [
+      Keying.config({
+        mode: 'special',
+        onEscape: (toolbar) => {
+          onClose(button, toolbar, ariaControls);
+          return Optional.some<boolean>(true);
+        }
+      }),
+      Sandboxing.config({
+        isPartOf: (container, data, queryElem): boolean => {
```

Comment:
> Is there absolutely no way to let these callbacks know of the skipFocus option from the external API?

These are all bad options, but they exist :)

a) mark the DOM in some way temporarily, that is removed after the function call
b) store the state in some way that is removed after the function call

The problem with approaches like this is that it's very easy for the "remove" part not to fire, and then they just stay in their "won't transfer focus state" any more. Having said that, maybe we could just explicitly remove them again when interacting with the keyboard. It might be the best bad option.

Reply:
@ephox-mogran Updated the solution by storing a `shouldSkipFocus` boolean in a singleton before `onOpen` and `onClose` callbacks are called. This works since the callbacks are synchronous. This singleton value is cleared after the call. Let me know what you think! :)